src/FSAL/FSAL_CORTXFS/mds.c

```diff
@@ -269,25 +280,146 @@ int kvsfs_pmds_ini(struct kvsfs_fsal_module *kvsfs,
 		goto out;
 	}
 
-	mds_ctx->pnfs_role = KVSFS_PNFS_DISABLED;
-	mds_ctx->mds = gsh_calloc(1, sizeof(struct kvsfs_pnfs_mds_info));
-	if (mds_ctx->mds == NULL) {
-		LogCrit(COMPONENT_PNFS,
+
+	/** 
+	 * TODO: When the pNFS config is moved to cortx.conf file, process
+	 * kvsfs_params here and update kvsfs->pnfs_role and
+	 * kvsfs->mds_ctx. Until then, just initialize kvsfs->mds_ctx.
+	 * In future, initialize only if pnfs_role > KVSFS_PNFS_DS.
+	 */
+	/* Read the pNFS.conf file */
+	rc = config_from_file("libcortxfs", CONF_FILE, &cfg_items,INI_STOP_ON_ERROR, &errors);
+	if(rc) {
+			free_ini_config_errors(errors);
+			rc = -rc;
+	} 
+
+	/* Read the configuration parameters */
+	rc = get_config_item("pNFS","pNFS_role",cfg_items, &items);
+	if(items==NULL) {
+		LogCrit(COMPONENT_PNFS,"pNFS parameters configured incorrectly\n");
+		rc = -rc;
+		goto out; 
+	}
+	pNFS_role = get_string_config_value(items, NULL);
```

Comment:
add check for pNFS_role == NULL

Reply:
Checking for items == NULL should take care of it right?